src/cpu/aarch64/acl_thread.cpp

```diff
@@ -65,8 +79,37 @@ void acl_set_threadpool_num_threads() {
         });
     }
 }
+// Swap BenchmarkScheduler for custom scheduler builds (i.e. ThreadPoolScheduler)
+void acl_set_benchmark_scheduler_tp() {
+    static std::once_flag flag_once;
+    // Create threadpool scheduler
+    std::unique_ptr<arm_compute::IScheduler> threadpool_scheduler
+            = std::make_unique<ThreadpoolScheduler>();
+    arm_compute::IScheduler *_real_scheduler = nullptr;
+    _real_scheduler = threadpool_scheduler.release();
+    // Create benchmark scheduler and set TP as real scheduler
+    std::shared_ptr<arm_compute::IScheduler> benchmark_scheduler
+            = std::make_unique<BenchmarkScheduler>(*_real_scheduler);
+    std::call_once(flag_once,
+            [&]() { arm_compute::Scheduler::set(benchmark_scheduler); });
+}
 #endif
 
+void set_acl_threading() {
+#if DNNL_CPU_THREADING_RUNTIME == DNNL_RUNTIME_OMP
+    acl_thread_bind();
+    if (verbose_has_create_profile()) { acl_set_benchmark_scheduler_default(); }
```

Comment:
IIUC, this benchmark scheduler adds prints for execution, not creation right?
If so, [verbose_has_exec_profile](https://github.com/oneapi-src/oneDNN/blob/f007d567226809f30546e2f15319066d4b124431/src/common/verbose.hpp#L147) should be used here.

Reply:
Hi, 
The reason I used create profile, because in the original implementation we wanted it for onednn_verbose setting of 2 and higher. 
Is there a way so that we can have it for verbose 2 or greater? So that we can use the timing measures from verbose 1 for performance analysis and keep verbose 2 when we want finer grained ACL information?

Reply:
Hi. We would typically like to avoid this kind of prints at verbose level 2.
For this you can use `debuginfo` levels and [verbose_debuginfo()](https://github.com/oneapi-src/oneDNN/blob/f007d567226809f30546e2f15319066d4b124431/src/common/verbose.hpp#L149).

On user side, `ONEDNN_VERBOSE=debuginfo=<level>` would need to be passed.

Reply:
Unfortunately this option does not work for us 
```
DNNL_VERBOSE=debuginfo=2 ./examples/primitives-convolution-cpp 
onednn_verbose,info,oneDNN v3.1.0 (commit a71acfecf09c79ea688b935f90dc3734b503029f)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:32
onednn_verbose,info,cpu,isa:AArch64 SVE (256bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,CpuPermuteKernel,0.0310059
onednn_verbose,exec,CpuGemmAssemblyWrapperKernel/a64_hybrid_fp32_mla_6x16,0.108154
Example passed on CPU.
Example passed on CPU.
```
As you can see, I can't tie the oneDNN primitive information to the underlying ACL kernel called for that primitive. It now just shows me the ACL information without the corresponding high-level ACL primitive.

Reply:
- Ideally, we want to turn on ACL-specific information with verbose>=2 because we would like to keep verbose=1 without ACL-specific information.
- We need a way to tie ACL kernel called to the oneDNN primitive and ONEDNN_VERBOSE=debuginfo=<level> does not work as mentioned above
- We also want this for release builds so that we can do measure time and compare results to running full stack models in frameworks.

Reply:
> Could you clarify why you want it with verbose level 2? What is the issue with ONEDNN_VERBOSE=debuginfo or ONEDNN_VERBOSE=profile_exec ?

We want to retain the original verbose 1 level without acl-specific information hence we don't want to have it with ONEDNN_VERBOSE=profile_exec (or onednn_verbose=1) . The issue with ONEDNN_VERBOSE=debuginfo is that it gives ACL-specific information, but I could add pd->info() to get the primitive-level information, then the information I am missing is the primitive-level `duration` that I can get from primitive_iface  https://github.com/oneapi-src/oneDNN/blob/73ec3de65693929c1089ce225b37dff6bd71dffa/src/common/primitive_iface.cpp#L108 

I was thinking of adding a new profile category acl_profile = 1 << 8 here https://github.com/oneapi-src/oneDNN/blob/f007d567226809f30546e2f15319066d4b124431/src/common/verbose.hpp#L121, would that work?
I want to get both onednn primitive information (pd->info() and duration_ms) AND (acl kernel + time)

Reply:
> I want to get both onednn primitive information (pd->info() and duration_ms) AND (acl kernel + time)

Verbose flags are cumulative. So you could pass `ONEDNN_VERBOSE=profile,debuginfo=2`, and it will print both primitive level timing information AND acl specific kernel timings.

> I was thinking of adding a new profile category acl_profile = 1 << 8 here

I would prefer something more general (e.g. `profile_external` as mentioned above) that could be reused for other similar usage where we want to profile external libraries in general (e.g. cudnn, miopen, ...).

Reply:
I have done the changes as per your suggestions. Thanks!